src/rust/ensogl/src/system/gpu/data/texture/class.rs

```diff
@@ -161,16 +283,13 @@ where S:StorageRelation<I,T> {
         let storage    = storage.into();
         let context    = context.clone();
         let gl_texture = context.create_texture().unwrap();
-        Self {storage,gl_texture,context}
+        let parameters = default();
+        Self {storage,gl_texture,context,parameters}
     }
 
     /// Sets the texture wrapping parameters.
-    pub fn set_texture_parameters(context:&Context) {
-        let target = Context::TEXTURE_2D;
-        let wrap   = Context::CLAMP_TO_EDGE as i32;
-        context.tex_parameteri(target,Context::TEXTURE_MIN_FILTER,Context::LINEAR as i32);
-        context.tex_parameteri(target,Context::TEXTURE_WRAP_S,wrap);
-        context.tex_parameteri(target,Context::TEXTURE_WRAP_T,wrap);
+    pub fn set_texture_parameters(&self, context:&Context) {
+        self.parameters.set_parameters(context);
```

Comment:
Something is bad with naming here. We have functions `Texture::set_parameters` and `Texture::set_texture_parameters`. Moreover the later is defined as `self.parameters.set_parameters(context);` which is totally not understandable. If it was meant to apply the parameters to context maybe using `apply_parameters` would make more sense?